### PR TITLE
[O11y][SQL Input] Add system test cases

### LINKED ISSUE #6285: Add System Tests for SQL Input Package
Add System Tests for SQL Input Package

### PR DESCRIPTION
**Pull request by aliabbas-elastic**
- Enhancement

## What does this PR do?

- This PR adds system test cases for the SQL Input package.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

## How to test this PR locally

- Clone integrations repo.
- Install elastic-package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/sql_input directory.
- Run the following command to run the system tests. elastic-package test system

## Related issues

- Closes #6285 

image

### CODE REVIEW

**Review by harnish-elastic — APPROVED** [review]
LGTM!

File: packages/sql_input/_dev/deploy/docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+version: '2.3'
+services:
+  sql_input:
+    image: mysql:8.0.13
```
Comment (lalit-satapathy) [inline packages/sql_input/_dev/deploy/docker/docker-compose.yml:4]: Lets add for couple of more drivers like: MSSQL and PostgreSQL

File: packages/sql_input/_dev/test/system/test-default-config.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+vars:
+  hosts:
+    - root:test@tcp({{Hostname}}:{{Port}})/
+  sql_queries: "- query: SHOW STATUS LIKE '%Threads%'\n  response_format: variables\n        \n"
```
Comment (lalit-satapathy) [inline packages/sql_input/_dev/test/system/test-default-config.yml:4]: Lets add for couple of more use cases. different response_format and other scenarios.

File: packages/sql_input/_dev/deploy/docker/docker-compose.yml (hunk 1 above)
Comment (aliabbas-elastic) [reply packages/sql_input/_dev/deploy/docker/docker-compose.yml:4]: Done

File: packages/sql_input/_dev/test/system/test-default-config.yml (hunk 2 above)
Comment (aliabbas-elastic) [reply packages/sql_input/_dev/test/system/test-default-config.yml]: According to the [sql](https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/sql/_meta/docs.asciidoc#:~:text=Expects%20sql%20query.-,response_format,-Either%20variables%20or) metricbeat module, `response_format` has two values. I have included them in the config files.

File: packages/sql_input/fields/input.yml [hunk 3]
```diff
@@ -2,3 +2,12 @@
   type: constant_keyword
   description: Sample field to be added.
   value: sql_metrics
+- name: sql.driver
```
Comment (lalit-satapathy) [inline packages/sql_input/fields/input.yml:5]: @ishleenk17 to check field mapping provided for input package

File: packages/sql_input/fields/input.yml [hunk 4]
```diff
@@ -2,3 +2,12 @@
...
+  type: keyword
+  description: The driver for which you want to run the queries.
+- name: sql.query
+  type: keyword
+  description: Query(s) that is/are to be executed.
+- name: sql.metrics.*
```
[the remark this answers is not in the document]
Comment (jsoriano) [reply packages/sql_input/fields/input.yml:11]: > For Input packages, we want the mappings to be done by user as per their convenience.

I think we have to provide working defaults in input packages. Of course users can add their custom mappings to make better use of their data with better knowledge of their use case.

In this case, the SQL module in Metricbeat has [mappings for these fields](https://github.com/elastic/beats/blob/36aa8847c0073abe5744595bc055a0bf07e6656c/x-pack/metricbeat/module/sql/_meta/fields.yml#L18), I would use the same definitions here.

Good to see that adding tests helped to catch up these missing fields :+1:

[the remark this answers is not in the document]
Comment (ishleenk17) [reply packages/sql_input/fields/input.yml:11]: @aliabbas-elastic : We can use the mappings which we have in the SQL module

[the remark this answers is not in the document]
Comment (aliabbas-elastic) [reply packages/sql_input/fields/input.yml:11]: @ishleenk17 Sure I can add the mappings according the SQL module. But also we have to keep this mapping.
```
- name: sql.metrics.*
  description: Default mapping for metric fields. You need to add your own custom mappings when storing other kinds of information.
  type: object
```
As I am receiving the metrics directly under the `metrics` object. Please refer the [sample_event.json](https://github.com/elastic/integrations/pull/6488/files#diff-c0c3589241b094f9805a20d0036fec6d5c4b05dedb594cdda264c5328f343c39R62).

[the remark this answers is not in the document]
Comment (aliabbas-elastic) [reply packages/sql_input/fields/input.yml:11]: @ishleenk17 Just to confirm its `sql.metrics.numeric.*` and it is already covered in the `sql.metrics.*`.

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**